src/TRestAxionGenericOptics.cxx

```diff
@@ -0,0 +1,166 @@
+///
+/// RESTsoft - Software for Rare Event Searches with TPCs
+///
+/// History of developments:
+///
+/// 2022-February: First concept and implementation of TRestAxionGenericOptics class.
+///            	  Johanna von Oy
+///
+/// \class      TRestAxionGenericOptics
+/// \author     Johanna von Oy <vonoy@physik.uni-bonn.de>
+///
+/// <hr>
+///
+
+#include "TRestAxionGenericOptics.h"
+
+using namespace std;
+#include <cmath>
+#include "TRestPhysics.h"
+using namespace REST_Physics;
+ClassImp(TRestAxionGenericOptics);
+
+///////////////////////////////////////////////
+/// \brief Default constructor
+///
+TRestAxionGenericOptics::TRestAxionGenericOptics() : TRestAxionOptics() { Initialize(); }
+
+///////////////////////////////////////////////
+/// \brief Default destructor
+///
+TRestAxionGenericOptics::~TRestAxionGenericOptics() {}
+
+///////////////////////////////////////////////
+/// \brief Constructor loading data from a config file
+///
+/// If no configuration path is defined using TRestMetadata::SetConfigFilePath
+/// the path to the config file must be specified using full path, absolute or
+/// relative.
+///
+/// The default behaviour is that the config file must be specified with
+/// full path, absolute or relative.
+///
+/// \param cfgFileName A const char* giving the path to an RML file.
+/// \param name The name of the specific metadata. It will be used to find the
+/// corresponding TRestAxionMagneticField section inside the RML.
+///
+TRestAxionGenericOptics::TRestAxionGenericOptics(const char* cfgFileName, string name)
+    : TRestAxionOptics(cfgFileName) {
+    debug << "Entering TRestAxionGenericOptics constructor( cfgFileName, name )" << endl;
+
+    Initialize();
+
+    LoadConfigFromFile(fConfigFileName, name);
+
+    if (GetVerboseLevel() >= REST_Info) PrintMetadata();
+}
+
+///////////////////////////////////////////////
+/// \brief Initialization of TRestAxionGenericOptics members
+///
+void TRestAxionGenericOptics::Initialize() {
+    TRestAxionOptics::Initialize();
+
+    SetSectionName(this->ClassName());
+    SetLibraryVersion(LIBRARY_VERSION);
+
+    // Call any initialization methods here
+}
+
+///////////////////////////////////////////////
+/// \brief Initialization of TRestAxionGenericOptics field members through a RML file
+///
+void TRestAxionGenericOptics::InitFromConfigFile() {
+    TRestAxionOptics::InitFromConfigFile();
+
+    /// TODO Initialize metadata members of this class here
+
+    // If we recover the metadata class from ROOT file we will need to call Initialize ourselves
+    this->Initialize();
+}
+
+///////////////////////////////////////////////
+/// \brief This calculates the interaction Point between the X-Ray photons path and a mirror in the
+/// predetermined layer
+///
+TVector3 TRestAxionGenericOptics::GetInteractionPoint(const TVector3& pos, const TVector3& dir) {
+    Int_t layer = TRestAxionOptics::GetEntranceShell(pos, dir);
```

Comment:
Line 141 looks strange. There is no need to call specifying the base class, mainly because the method `GetEntranceShell` is not virtual, and it is not re-implemented inside `TRestAxionGenericOptics`. Therefore, just calling `GetEntranceShell` will have the same effect. When the method might be existing on both, base and inherited, then we need to specify, and therefore to me it looks as if `GetEntrancePoint` is on both classes.

Reply:
Ah okay, I understand, thanks!